lib/dpl/provider/elastic_beanstalk.rb

```diff
@@ -88,7 +92,13 @@ def create_zip
       end
 
       def upload(key, file)
-        obj = s3.buckets[bucket_name].objects[key]
+        obj = s3.buckets[bucket_name]
+        file_path = bucket_path
+        if file_path
+          file_path << '/' unless file_path.end_with?('/')
```

Comment:
> This combination of if and unless doesn't read well.
>
> `file_path.gsub!(/\/*$/,'/')` should be sufficient.

Fair point. I'll change it up.

> Also, what is the reason for file_path = bucket_path? I don't think bucket_path is used elsewhere, and just having file_path pointing to the same object as bucket_path does not make sense to me.

The reason that I created a `bucket_path` variable was to attempt to stay in line with the provider itself. I noticed that there were methods created for every option that could be passed in.

I could do something like

```ruby
path = options[:bucket_path]
if path
    path.gsub!(/\/*$/,'/')
end

obj = obj.objects["#{path}#{key}"]
obj.write(Pathname.new(file))
obj
```

instead. Or were you talking about modifying the value inside of the `options` hash?